lib/users/routes.js

```diff
@@ -30,18 +35,203 @@ const all = (req, res, next) => {
+      });
+    });
+  }
+
+  // grab the application from the database
+  return database('users').where({
+    id: req.params.id,
+  }).then(rows => {
+    // application not in database; send to 404
+    if (rows.length < 1) {
+      const err = {
+        message: config.users.messages.missing.id.replace('%id', req.params.id),
+        safe: `/${config.users.base}`,
+        status: 404,
+      };
+
+      return next(err);
+    }
+
+    const user = rows[0];
+
+    // add session data for this application
+    _.set(req.session, 'form.users.edit', {
+      id: user.id,
+    });
+
+    const data = {
+      'email': {
+        'email': {
+          'value': user.email,
+        },
+      },
+      'role': {
+        'select': {
+          'value': user.role,
+        },
+      },
+    };
+
+    return types.only('users', data, [merged], config);
+  }).then(only => {
+    return types.form(only, null, config);
+  }).then(form => {
+    res.render('users/one', {
+      form,
+      action: `/${config.users.base}/${config.users.actions.save}`,
+      config: config.users,
+      button: config.users.actions.update,
+    });
+
+
+    return true;
+  }).catch(e => {
+    const err = {
+      message: e.message,
+      safe: `/${config.users.base}`,
+      status: 404,
+    };
+
+    return next(err);
+  });
+};
 
 /**
  * Save/delete a user
+ * @param {object} req - HTTP Request
+ * @param {object} res - HTTP Response
+ *
+ * @returns {promise} database promise
  */
-const save = () => {};
+const save = (req, res) => {
+  const id = _.get(req.session, 'form.users.edit.id', null);
+  const referrer = (_.get(req.session, 'referrer') || req.get('Referrer')) || `/${config.users.base}`;
+  const merged = req.app.get('users-model') || {};
+
+  // user hit delete button
+  if (req.body.submit === config.users.actions.delete && id) {
+    return database('users').where('id', '=', id).del().then(() => {
+      _.unset(req.session, 'form.users.edit');
+      _.unset(req.session, 'form.users.save');
+
+      res.redirect(`/${config.users.base}`);
+
+      return true;
+    });
+  }
+
+  // Validation
+  const validated = types.form.validate(req.body, merged, 'save');
+
+  if (validated === true) {
+    const data = {
+      'email': req.body['email--email'],
+      'role': req.body['role--select'],
+      'password': bcrypt.hashSync(req.body['password--password']),
+    };
+
+    // `update` is for the Edit form
+    if (req.body.submit === config.users.actions.update && id) {
+      return database('users').where('id', '=', id).update(data).returning('*').then(() => {
+        _.unset(req.session, 'form.users.edit');
+        _.unset(req.session, 'form.users.save');
+
+        res.redirect(`/${config.users.base}`);
+
+        return true;
+      });
+    }
+
+    return database('users').insert(data).returning('*').then(added => {
+      _.unset(req.session, 'form.users.save');
+
+      res.redirect(`/${config.users.base}/${added[0].id}`);
```

Comment:
Why does creating a new user redirect to the individual user's page instead of the user listing page? There is no further action to take on the individual user page.